### DISCUSSION TITLE
how to repeat the optimization of the vertical coupler in this article？

### DISCUSSION THREAD

**Discussion by wllshr123** [NanoComp/meep#2393] [Q&A]
How should I reproduce this article correctly?
“Multi-layer inverse design of vertical grating couplers for high-density, commercial foundry interconnects”
I want to repeat the optimization of the vertical coupler in the article, but since I am a new user of meep, I would like to ask you to introduce the optimization steps to me, or would you like to give me a complete code so that I can try to run it?

**Comment by smartalecH** [comment]
Glad you enjoyed our paper. 

You can find several adjoint optimization tutorials in the form of interactive notebooks here: https://github.com/NanoComp/meep/tree/master/python/examples/adjoint_optimization

Unfortunately, we don't have a tutorial with the exact code used to produce the results of that paper. Those designs were fabricated on a commercial foundry platform, and as such contain sensitive proprietary details. 

That being said, we've been planning to do a series of grating coupler tutorials (for both 2D and 3D designs).

(If anyone else is interested in writing a tutorial, that would also be great!)

**Reply by wllshr123** [reply]
Thank you for your reply.In addition to the content in this link, do you have any other tutorials with adjoint optimization? (preferably 3D)

**Reply by smartalecH** [reply]
You can also take a look at the following repo:

https://github.com/NanoComp/photonics-opt-testbed

These, along with the tutorials we mentioned earlier are all the public examples we have (for now).

**Reply by XiangFFF** [reply]
> Glad you enjoyed our paper.
> 
> You can find several adjoint optimization tutorials in the form of interactive notebooks here: https://github.com/NanoComp/meep/tree/master/python/examples/adjoint_optimization
> 
> Unfortunately, we don't have a tutorial with the exact code used to produce the results of that paper. Those designs were fabricated on a commercial foundry platform, and as such contain sensitive proprietary details.
> 
> That being said, we've been planning to do a series of grating coupler tutorials (for both 2D and 3D designs).
> 
> (If anyone else is interested in writing a tutorial, that would also be great!)

Hello, I'm currently working on grating coupler design and interested in the tutorial. Has there been any recent progress on it? I'd be glad to contribute to the relevant tutorial content or assist with code testing if possible.
